src/components/DisplaySetting.js

```diff
@@ -1,42 +1,43 @@
 import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
 import "../App.css";
+
 function DisplaySetting(props) {
-  // const [displaySettingComponent, setDisplaySettingComponent] = useState("none");
   const { attributesName, handleOutputImagesAttributes } = props;
-  // const [attributesToHide, setAttributesToHide] = useState([]);
-
   const [attributesToShow, setAttributesToShow] = useState(attributesName);
-  const handleDisplaySettingChange = (event) => {
-    const { name, value } = event.target;
+  const [tempAttributes, setTempAttributes] = useState(attributesName);
 
-    attributesToShow.includes(name) === true
-      ? setAttributesToShow(
-          attributesToShow.filter((attribute) => {
-            if (attribute === name) return false;
-            else return true;
-          })
-        )
-      : setAttributesToShow([...attributesToShow, name]);
+  const setAttributesToShowHelper = (name) => {
+    if (attributesToShow.includes(name)) {
+      return attributesToShow.filter((attribute) => {
+        if (attribute === name) return false;
+        return true;
+      });
+    }
+    return [...attributesToShow, name];
+  };
+  const handleDisplaySettingChange = (event) => {
+    const { name } = event.target;
+    setAttributesToShow(setAttributesToShowHelper(name));
 
-    console.log("value=", value);
-    console.log("name=", name);
-    console.log("in DisplaySetting attributesToShow=", attributesToShow);
-    //   console.log("displaySettingComponent is=", displaySettingComponent);
+    setTempAttributes(
```

Comment:
This tempAttribute is rendering the columns value to show one step slower then the program